modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -423,14 +436,32 @@ def _compute_axis_labels(self, axis: int, partitions=None):
         -------
         pandas.Index
             Labels for the specified `axis`.
+        List of int
+            Size of partitions alongsize specified `axis`.
         """
         if partitions is None:
             partitions = self._partitions
-        new_index, _ = self._partition_mgr_cls.get_indices(axis, partitions)
-        return new_index
+        new_index, internal_idx = self._partition_mgr_cls.get_indices(axis, partitions)
+        return new_index, list(map(len, internal_idx))
+
+    def _filter_empties(self, compute_lengths=True):
+        """
+        Remove empty partitions from `self._partitions` to avoid triggering excess computation.
+
+        Parameters
+        ----------
+        compute_lengths : bool, default: True
+            Trigger the computations for partition sizes if they're not done already.
+        """
+        if not compute_lengths and (
+            self._index_cache is None
+            or self._columns_cache is None
+            or self._row_lengths_cache is None
+            or self._column_widths_cache is None
```

Comment:
```suggestion
            self._index_cache is not None
            or self._columns_cache is not None
            or self._row_lengths_cache is not None
            or self._column_widths_cache is not None
```
Should this be so?

Reply:
No, the idea is expressed right - if we don't know some of the caches, and we're requested to _not_ compute we do not go further, this is to delay the blocking on partitions as much as possible.

Reply:
Why do we need to use `self._index_cache` and `self._columns_cache` in the condition if the docstring for `compute_lengths` says about partition sizes?

Reply:
renamed the parameter to `compute_metadata` to align its naming with the `lazy_metadata_decorator()`